daphne/src/vdaf/prio3.rs

```diff
@@ -60,6 +60,12 @@ pub(crate) fn prio3_shard(
             let vdaf = Prio3::new_sum(2, *bits)?;
             Ok(shard!(vdaf, &(measurement as u128), nonce))
         }
+        (Prio3Config::SumVec { bits, len }, DapMeasurement::U64Vec(measurement)) => {
+            let vdaf = Prio3::new_sum_vec(2, *bits, *len)?;
+            // TODO(tholop): shard expects a u128. Is it ok to allocate a new vector to cast each element?
```

Comment:
I added a new `DapMeasurement` variant, for which we don't need this allocation. But do we still want to keep the U64 type? I used it originally because that's what `Count` and `Sum` are using (and they cast to u128).

Reply:
Hmm, I don't think `Count` casts to `u128`: https://github.com/cloudflare/daphne/pull/331/files#diff-6660c5f7b70da1bf7dd0a22e00b5ee3191e693b9cef73a0fb5aeb31ae514f28cR53

Good point about `Sum`, however! (Also true for `Histogram`.) I don't remember the original motivation for this (or if this was well thought out), but I think it makes sense to use the measurement type that's native for the `Type`, i.e.:
* `Count` should be `u64`
* `Histogram`, `Sum` should be `u128`
* `SumVec` should be `Vec<u128` (in particular we should not define a `U64Vec` variant)

However to keep this change minimal, I'd suggest we create another PR to fix `Count` and `Histogram`. (Alternatively, we could file an issue and address it later.)